analyzers/src/SonarAnalyzer.CSharp/Rules/DisposableNotDisposed.cs

```diff
@@ -40,7 +40,13 @@ public sealed class DisposableNotDisposed : SonarDiagnosticAnalyzer
                 KnownType.System_Net_Sockets_TcpClient,
                 KnownType.System_Net_Sockets_UdpClient);
 
-        private static readonly ISet<string> DisposeMethods = new HashSet<string> { "Dispose", "Close" };
+        private static readonly ImmutableArray<KnownType> DisposableTypes = ImmutableArray.Create
+        (
+            KnownType.System_IDisposable,
+            KnownType.System_IAsyncDisposable
+        );
+
+        private static readonly ISet<string> DisposeMethods = new HashSet<string> { "Dispose", "DisposeAsync", "Close" };
 
         private static readonly ISet<string> FactoryMethods = new HashSet<string>
         {
```

Comment:
(I **suggest to get this PR merged without these naming changes** from this comment, and do the naming changes immediately after, separate PR & master commit, as we both have the context)

clean as you code observations on improving the names inside `Initialize` - these are my suggestions, I'm happy to find better names:

- `namedTypeSymbol` --> `namedType` (less keystrokes; this is how we name the parameter in `TrackInitializedLocalsAndPrivateFields` so it's already the pattern in the file in other places)
-  `trackedNodesAndSymbols` --> `disposableObjects` (it's kinda obvious that we're tracking them)
- `excludedSymbols` --> `probablyDisposed` (does this sound better in your opinion?)

Reply:
another not-for-this-PR observation: we only track classes and structs in this class

why not also record? maybe open should be a separate ticket to support records?

Reply:
Noted. Records are covered by the C# 9 test cases (DisposableNotDisposed.CSharp9.cs).